File: Assemblies/ProteanCMS/CMS/Cart/PaymentProviders/SkipPayment.vb
Imports System.Xml
Imports System.Web.HttpUtility
Imports System.Web.Configuration
Imports System.IO
Imports System.Collections.Generic
Imports System.Data
Imports System.Data.SqlClient
Imports VB = Microsoft.VisualBasic
Imports Protean.Cms.Cart
Imports System.Net
Imports CardinalCommerce
Imports System.IdentityModel.Tokens
Imports System.IdentityModel.Tokens.Jwt
Imports System.Security.Claims
Imports System.Text
Imports Newtonsoft.Json.Linq


Namespace Providers
    Namespace Payment
        Public Class SkipPayment

            Private Const mcModuleName As String = "Providers.Payment.SkipPayment"

            Public Sub New()
                'do nothing
            End Sub

            Public Sub Initiate(ByRef _AdminXforms As Object, ByRef _AdminProcess As Object, ByRef _Activities As Object, ByRef PayProvider As Object, ByRef myWeb As Cms)
                Dim cProcessInfo As String = ""
                Try

                    PayProvider.AdminXforms = New AdminXForms(myWeb)
                    PayProvider.AdminProcess = New AdminProcess(myWeb)
                    '   PayProvider.AdminProcess.oAdXfm = PayProvider.AdminXforms
                    PayProvider.Activities = New Activities()

                Catch ex As Exception
                    returnException(myWeb.msException, mcModuleName, "Initiate", ex, "", cProcessInfo, gbDebug)

                End Try
            End Sub

            Public Class AdminXForms
                Inherits Cms.Admin.AdminXforms
                Private Const mcModuleName As String = "Providers.Providers.Eonic.AdminXForms"

                Sub New(ByRef aWeb As Cms)
                    MyBase.New(aWeb)
                End Sub

            End Class

            Public Class AdminProcess
                Inherits Cms.Admin

                Dim _oAdXfm As Protean.Providers.Payment.PayPalPro.AdminXForms

                Public Property oAdXfm() As Object
                    Set(ByVal value As Object)
                        _oAdXfm = value
                    End Set
                    Get
                        Return _oAdXfm
                    End Get
                End Property

                Sub New(ByRef aWeb As Cms)
                    MyBase.New(aWeb)
                End Sub
            End Class


            Public Class Activities

                Private Const mcModuleName As String = "Providers.Payment.PayPalPro.Activities"
                Private myWeb As Protean.Cms
                Protected moPaymentCfg As XmlNode
                Private nTransactionMode As TransactionMode

                Enum TransactionMode
                    Live = 0
                    Test = 1
                    Fail = 2
                End Enum


                Public Function CollectPayment(ByRef oWeb As Protean.Cms, ByVal nPaymentMethodId As Long, ByVal Amount As Double, ByVal CurrencyCode As String, ByVal PaymentDescription As String, ByRef oCart As Protean.Cms.Cart) As String
                    Dim cProcessInfo As String = ""
                    Try

                        'Do nothing because recurring payments are automatic

                        Return "Success"

                    Catch ex As Exception
                        returnException(myWeb.msException, mcModuleName, "CollectPayment", ex, "", cProcessInfo, gbDebug)
                        Return "Payment Error"
                    End Try
                End Function

                Public Function GetPaymentForm(ByRef myWeb As Protean.Cms, ByRef oCart As Cms.Cart, ByRef oOrder As XmlElement, Optional returnCmd As String = "cartCmd=SubmitPaymentDetails") As Protean.xForm
                    PerfMon.Log("Protean.Providers.payment.SkipPayment", "GetPaymentForm")
                    Dim cProcessInfo As String = ""
                    Try

                        Dim ccXform As xForm = New Protean.xForm(myWeb.moCtx)

                        ccXform.NewFrm("SkipPayment")

                        'Dim ccXform As New Protean.xForm(myWeb.moCtx)
                        ccXform.valid = True
                        Return ccXform

                    Catch ex As Exception
                        returnException(myWeb.msException, mcModuleName, "GetPaymentForm", ex, "", cProcessInfo, gbDebug)
                        Return Nothing
                    End Try
                End Function




                Public Function CheckStatus(ByRef oWeb As Protean.Cms, ByRef nPaymentProviderRef As String) As String
                    Dim cProcessInfo As String = ""
                    Try

                        Return "Error - no value returned"


                    Catch ex As Exception
                        returnException(myWeb.msException, mcModuleName, "CheckStatus", ex, "", cProcessInfo, gbDebug)
                        Return ""
                    End Try

                End Function

                Public Function CancelPayments(ByRef oWeb As Protean.Cms, ByRef nPaymentProviderRef As String) As String
                    Dim cProcessInfo As String = ""

                    Try

                        Return CheckStatus(oWeb, nPaymentProviderRef)

                    Catch ex As Exception
                        returnException(myWeb.msException, mcModuleName, "CancelPayments", ex, "", cProcessInfo, gbDebug)
                        Return ""
                    End Try

                End Function

                Public Function AddPaymentButton(ByRef oOptXform As xForm, ByRef oFrmElmt As XmlElement, ByVal configXml As XmlElement, ByVal nPaymentAmount As Double, ByVal submissionValue As String, ByVal refValue As String) As Boolean

                    Try

                        Dim nMaxAmt As Double = CDbl("0" & configXml.SelectSingleNode("MaxValue").InnerText)
                        Dim nMinAmt As Double = CDbl("0" & configXml.SelectSingleNode("MinValue").InnerText)

                        If nMaxAmt <= nPaymentAmount And nMinAmt >= nPaymentAmount Then

                            Dim PaymentLabel As String = configXml.SelectSingleNode("description/@value").InnerText
                            'allow html in description node...
                            Dim bXmlLabel As Boolean = False

                            If configXml.SelectSingleNode("description").InnerXml <> "" Then
                                PaymentLabel = configXml.SelectSingleNode("description").InnerXml
                                bXmlLabel = True
                            End If

                            Dim iconclass As String = ""
                            If Not configXml.SelectSingleNode("icon/@value") Is Nothing Then
                                iconclass = configXml.SelectSingleNode("icon/@value").InnerText
                            End If

                            oOptXform.addSubmit(oFrmElmt, submissionValue, PaymentLabel, refValue, "pay-button pay-" & configXml.GetAttribute("name"), iconclass, configXml.GetAttribute("name"))

                        End If

                    Catch ex As Exception

                    End Try
                End Function


            End Class
        End Class
    End Namespace
End Namespace

File: Assemblies/Protean.CMS/features/cart/PaymentProviders/SkipPayment.cs
﻿using Microsoft.VisualBasic.CompilerServices;
using System;
using System.Xml;
using static Protean.stdTools;


namespace Protean.Providers
{
    namespace Payment
    {
        public class SkipPayment : IPaymentProvider
        {

            private const string mcModuleName = "Providers.Payment.SkipPayment";

            public SkipPayment()
            {
                // do nothing
            }

            private IPaymentAdminXforms _AdminXforms;
            private IPaymentAdminProcess _AdminProcess;
            private IPaymentActivities _Activities;
            IPaymentAdminXforms IPaymentProvider.AdminXforms
            {
                set
                {
                    _AdminXforms = value;
                }
                get
                {
                    return _AdminXforms;
                }
            }
            IPaymentAdminProcess IPaymentProvider.AdminProcess
            {
                set
                {
                    _AdminProcess = value;
                }
                get
                {
                    return _AdminProcess;
                }
            }
            IPaymentActivities IPaymentProvider.Activities
            {
                set
                {
                    _Activities = value;
                }
                get
                {
                    return _Activities;
                }
            }
            public IPaymentProvider Initiate(ref Cms myWeb)
            {
                string cProcessInfo = "";
                try
                {

                    _AdminXforms = new AdminXForms(ref myWeb);
                    _AdminProcess = new AdminProcess(ref myWeb);
                    // MemProvider.AdminProcess.oAdXfm = MemProvider.AdminXforms
                    _Activities = new Activities();
                    return this;
                }

                catch (Exception ex)
                {
                    stdTools.returnException(ref myWeb.msException, mcModuleName, "Initiate", ex, "", cProcessInfo, gbDebug);
                    return this;
                }
            }

            public class AdminXForms : Cms.Admin.AdminXforms, IPaymentAdminXforms
            {
                private const string mcModuleName = "Providers.Providers.Eonic.AdminXForms";

                public AdminXForms(ref Cms aWeb) : base(ref aWeb)
                {
                }

            }

            public class AdminProcess : Cms.Admin, IPaymentAdminProcess
            {

                //private Protean.Providers.Payment.PayPalPro.AdminXForms _oAdXfm;

                public object oAdXfm
                {
                    set
                    {
                        // _oAdXfm = value;
                    }
                    get
                    {
                        // return _oAdXfm;
                        return "";
                    }
                }

                public AdminProcess(ref Cms aWeb) : base(ref aWeb)
                {
                }
            }


            public class Activities : IPaymentActivities
            {

                private const string mcModuleName = "Providers.Payment.PayPalPro.Activities";
                private Cms myWeb;
                protected XmlNode moPaymentCfg;
                //private TransactionMode nTransactionMode;

                public enum TransactionMode
                {
                    Live = 0,
                    Test = 1,
                    Fail = 2
                }


                public string CollectPayment(ref Cms oWeb, long nPaymentMethodId, double Amount, string CurrencyCode, string PaymentDescription, ref Cms.Cart oCart)
                {
                    string cProcessInfo = "";
                    try
                    {

                        // Do nothing because recurring payments are automatic

                        return "Success";
                    }

                    catch (Exception ex)
                    {
                        stdTools.returnException(ref myWeb.msException, mcModuleName, "CollectPayment", ex, "", cProcessInfo, gbDebug);
                        return "Payment Error";
                    }
                }

                public Protean.xForm GetPaymentForm(ref Cms myWeb, ref Cms.Cart oCart, ref XmlElement oOrder, string returnCmd = "cartCmd=SubmitPaymentDetails")
                {
                    string cProcessInfo = "";
                    try
                    {

                        var ccXform = new Protean.xForm(myWeb.moCtx, ref myWeb.msException);

                        ccXform.NewFrm("SkipPayment");

                        // Dim ccXform As New Protean.xForm(myWeb.moCtx)
                        ccXform.valid = true;
                        return ccXform;
                    }

                    catch (Exception ex)
                    {
                        stdTools.returnException(ref myWeb.msException, mcModuleName, "GetPaymentForm", ex, "", cProcessInfo, gbDebug);
                        return (Protean.xForm)null;
                    }
                }




                public string CheckStatus(ref Cms oWeb, ref string nPaymentProviderRef)
                {
                    string cProcessInfo = "";
                    try
                    {

                        return "Error - no value returned";
                    }


                    catch (Exception ex)
                    {
                        stdTools.returnException(ref myWeb.msException, mcModuleName, "CheckStatus", ex, "", cProcessInfo, gbDebug);
                        return "";
                    }

                }

                public string CancelPayments(ref Cms oWeb, ref string nPaymentProviderRef)
                {
                    string cProcessInfo = "";

                    try
                    {

                        return CheckStatus(ref oWeb, ref nPaymentProviderRef);
                    }

                    catch (Exception ex)
                    {
                        stdTools.returnException(ref myWeb.msException, mcModuleName, "CancelPayments", ex, "", cProcessInfo, gbDebug);
                        return "";
                    }

                }

                public bool AddPaymentButton(ref Cms myWeb, ref Cms.Cart myCart, ref Cms.xForm oOptXform, ref XmlElement oFrmElmt, XmlElement configXml, double nPaymentAmount, string submissionValue, string refValue)
                {

                    try
                    {

                        double nMaxAmt = Conversions.ToDouble("0" + configXml.SelectSingleNode("MaxValue").InnerText);
                        double nMinAmt = Conversions.ToDouble("0" + configXml.SelectSingleNode("MinValue").InnerText);

                        if (nMaxAmt <= nPaymentAmount & nMinAmt >= nPaymentAmount)
                        {

                            string PaymentLabel = configXml.SelectSingleNode("description/@value").InnerText;
                            // allow html in description node...
                            // bool bXmlLabel = false;

                            if (!string.IsNullOrEmpty(configXml.SelectSingleNode("description").InnerXml))
                            {
                                PaymentLabel = configXml.SelectSingleNode("description").InnerXml;
                                // bXmlLabel = true;
                            }

                            string iconclass = "";
                            if (configXml.SelectSingleNode("icon/@value") != null)
                            {
                                iconclass = configXml.SelectSingleNode("icon/@value").InnerText;
                            }

                            oOptXform.addSubmit(ref oFrmElmt, submissionValue, PaymentLabel, refValue, "pay-button pay-" + configXml.GetAttribute("name"), iconclass, configXml.GetAttribute("name"));

                        }
                    }

                    catch (Exception)
                    {

                    }

                    return default;
                }

                public xForm GetRedirect3dsForm(ref Cms myWeb)
                {
                    throw new NotImplementedException();
                }

                public XmlElement UpdatePaymentStatus(ref Cms oWeb, ref long nPaymentMethodKey)
                {
                    throw new NotImplementedException();
                }

                public string GetMethodDetail(ref Cms oWeb, ref string nPaymentProviderRef)
                {
                    throw new NotImplementedException();
                }

                public void ValidatePaymentByCart(int nCartId, bool bValid)
                {
                    throw new NotImplementedException();
                }

                public string RefundPayment(string providerPaymentReference, decimal amount, string validGroup = "")
                {
                    throw new NotImplementedException();
                }

                public string UpdateOrderWithPaymentResponse(string AuthNumber, string validGroup = "")
                {
                    throw new NotImplementedException();
                }

                public string ProcessNewPayment(string orderId, decimal amount, string cardNumber, string cV2, string expiryDate, string startDate, string cardHolderName, string address1, string address2, string town, string postCode, string cCounty = "", string cCountry = "", string validGroup = "")
                {
                    throw new NotImplementedException();
                }

                public XmlElement GetWalletPaymentDetails(XmlElement opElemt)
                {
                    throw new NotImplementedException();
                }
            }
        }
    }
}